pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -107,8 +108,12 @@ func NewController(jobInformer batchv1informers.JobInformer, cronJobsInformer ba
 
 	jm := &Controller{
 		kubeClient: kubeClient,
-		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(DefaultCronJobBackOff, MaxCronJobBackOff), "cronjob"),
-		recorder:   eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cronjob-controller"}),
+		// you don't need this queue, it is used in jobs b/c we use that
+		// for retries, for cronjob we won't do that this way, regular
+		// delaying queue should be sufficient: NewNamedDelayingQueue is
+		// what you're looking for
+		queue:    workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(DefaultCronJobBackOff, MaxCronJobBackOff), "cronjob"),
```

Comment:
will refactor with DelayingQueue and requeue immediately in case of error instead of exponential backoff